src/CodeGen_LLVM.h

```diff
@@ -434,6 +438,11 @@ class CodeGen_LLVM : public IRVisitor {
                                         bool zero_initialize = false,
                                         const std::string &name = "");
 
+    /** Total size of all alloca() storage requested (including alignment padding).
```

Comment:
It will be substantially less, because allocas with non-overlapping lifetimes will share space. So in a pipeline with n stages doing stack allocations of size m, we'll see O(n*m) bytes allocated on the stack, but probably only use O(m) stack. Register promotions is a smaller effect.

So I suggest tweaking the comment to make it clear that this is a conservative upper bound.

Reply:
I think this isn't guaranteed to be an upper bound, there are probably other things that go on the stack that aren't being accounted for (function call overhead, stack spills from register allocation, probably other things).

Reply:
PTAL